### ISSUE TITLE
Linux Nvidia glXGetProcAddress Fail to compile

### ISSUE THREAD

**Issue by adderly** [openglsuperbible/sb6code#8]
Hi ,
Under ubuntu 13.04 with nvidia 331.xx i get this compilation error.

/usr/bin/ld: /usr/local/lib/libglfw.so: undefined reference to symbol 'glXGetProcAddress'
/usr/bin/ld: note: 'glXGetProcAddress' is defined in DSO /usr/lib/gcc/x86_64-linux-gnu/4.7/../../../x86_64-linux-gnu/libGL.so so try adding it to the linker command line.
/usr/lib/gcc/x86_64-linux-gnu/4.7/../../../x86_64-linux-gnu/libGL.so: could not read symbols: Invalid operation

Which i guess isn't supported in the nvidia stuff?
what workaround i could have?

**Comment by grahamsellers** [comment]
The samples do work on NVIDIA + Linux as far as I know. I will have to drop an NVIDIA board in my Linux machine and see what the problem is.

**Comment by bradc6** [comment]
Try using my fork @ https://github.com/bradc6/sb6code . I think I had the same issue as you (Ubuntu 12.04 Nvidia 331), it was due the compiler not looking in the right location for glfw.

**Comment by adderly** [comment]
@grahamsellers Yes, please. On AMD just works, look http://code.mythtv.org/trac/ticket/4064 . There is a solution there.
@bradc6 So, what did you changed in the cmake file?

**Comment by bradc6** [comment]
@adderly I changed it to use parts of Cmake system to find the location of gl and glfw via find_package() calls, if found then use the found libraries instead of the hard coded defaults. As soon as I did that everything worked

**Comment by adderly** [comment]
@bradc6 Im going to try it, when i get home. Hope it works, and what are the differences between the library included and the one the system have? Different Version?
Well last night i was messing with the packages, hopefully i didn't brake anything.

**Comment by grahamsellers** [comment]
I have some pending patches and pull requests to fix some CMake issues. I really wanted to test them before I commit. I'll try to get to it ASAP - hopefully that will resolve the issues.

**Comment by Xonar** [comment]
It works for me (NVIDIA). I had to change elif to elseif (I checked there is already a pull request for it) to make it work with my version of cmake,
